File: 2022/typescript/day01/solve.ts
const getElves = async (path: string) => {
  const f = await Deno.readTextFile(path);
  return f
    .split("\n\n")
    .map((elf) => elf.split("\n").map((cal) => Number(cal)));
};

const p1 = async () => {
  const elves = await getElves("input.txt");
  const totalCals = elves.map((e) => e.reduce((a, b) => a + b, 0));
  return Math.max(...totalCals);
};

const p2 = async () => {
  const elves = await getElves("input.txt");
  const totalCals = elves.map((e) => e.reduce((a, b) => a + b, 0));
  totalCals.sort((a, b) => b - a);
  return totalCals[0] + totalCals[1] + totalCals[2];
};

const part1 = await p1();
const part2 = await p2();
console.log("Part 1: ", part1);
console.log("Part 2: ", part2);

File: 2022/python/day01/solve.py
from functools import reduce

def get_data(path_to_data):
  with open(path_to_data, 'r') as f:
    groups = f.read().split('\n\n')
    elves = [list(map(int, g.split('\n'))) for g in groups]
  return elves

def p1():
  elves = get_data("input.txt")
  total_cals = [reduce(lambda a, b: a + b, e) for e in elves]
  total_cals.sort(reverse=True)
  print('Part 1: {}'.format(total_cals[0]))

def p2():
  elves = get_data("input.txt")
  total_cals = [reduce(lambda a, b: a + b, e) for e in elves]
  total_cals.sort(reverse=True)
  print('Part 2: {}'.format(total_cals[0] + total_cals[1] + total_cals[2]))

p1()
p2()